controllers/admin/AdminCartRulesController.php

```diff
@@ -671,7 +671,7 @@ public function renderForm()
                             $value = $carrier['id_carrier'] . ' - ' . Configuration::get('PS_SHOP_NAME');
                         } else {
                             $value = $carrier['id_carrier'] . ' - ' . $carrier['name'];
-                            if ($carrier['name']) {
+                            if ($carrier['delay']) {
```

Comment:
Just think about the delay, is it possible to have `0` as value?
Shouldn't be something like:
```suggestion
                            if (isset($carrier['delay'])) {
```